subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/AndroidStudioSystemHelper.java

```diff
@@ -1,32 +1,44 @@
 package org.gradle.profiler.studio.plugin.system;
 
-import com.android.tools.idea.gradle.project.sync.GradleSyncInvoker;
 import com.android.tools.idea.gradle.project.sync.GradleSyncState;
+import com.android.tools.idea.projectsystem.ProjectSystemSyncManager.SyncReason;
+import com.android.tools.idea.projectsystem.ProjectSystemUtil;
 import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.progress.ProgressIndicator;
 import com.intellij.openapi.project.Project;
+import com.intellij.openapi.startup.StartupManager;
 import com.intellij.openapi.wm.IdeFrame;
 import com.intellij.openapi.wm.ex.StatusBarEx;
 import com.intellij.openapi.wm.ex.WindowManagerEx;
 import com.intellij.util.messages.MessageBusConnection;
 import org.gradle.profiler.studio.plugin.system.GradleProfilerGradleSyncListener.GradleSyncResult;
 
-import static com.google.wireless.android.sdk.stats.GradleSyncStats.Trigger.TRIGGER_USER_SYNC_ACTION;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+
+import static com.android.tools.idea.projectsystem.ProjectSystemSyncUtil.PROJECT_SYSTEM_SYNC_TOPIC;
 
 public class AndroidStudioSystemHelper {
 
     private static final long WAIT_ON_PROCESS_SLEEP_TIME = 10;
+    private static final long WAIT_ON_STARTUP_SLEEP_TIME = 100;
 
     /**
      * Does a Gradle sync.
      */
     public static GradleSyncResult doGradleSync(Project project) {
+        MessageBusConnection connection = project.getMessageBus().connect();
         GradleProfilerGradleSyncListener syncListener = new GradleProfilerGradleSyncListener();
+        connection.subscribe(GradleSyncState.GRADLE_SYNC_TOPIC, syncListener);
         try {
```

Comment:
```suggestion
        MessageBusConnection connection = project.getMessageBus().connect();
        try {
            GradleProfilerGradleSyncListener syncListener = new GradleProfilerGradleSyncListener();
            connection.subscribe(GradleSyncState.GRADLE_SYNC_TOPIC, syncListener);
```

Reply:
We can't move `GradleProfilerGradleSyncListener syncListener = new GradleProfilerGradleSyncListener();` since it's used later, but we can move connection subscribe.

Reply:
I changed the code to call listener directly from `try {}`